src/generate_form.jsx

```diff
@@ -0,0 +1,285 @@
+					(<TextField
+						className={textfield_classes.textfield}
+        				id={props.metadata.name}
+						type="number"
+						onChange={handleChange}
+						error={props.err}
+        				margin="dense"
+        				width="auto"/>)
+					:
+					(<Select
+						className={select_classes.select}
+						value={val}
+						error={props.err}
+						onChange={handleChange}
+						inputProps={{
+							name: 'val',
+							id: props.metadata.name
+						}}>
+							{Array.from(Array(props.metadata.upper).keys()).map(
+								(num) => 
+								<MenuItem value={num+1} key={props.metadata.name + " Option " + num}>{num+1}</MenuItem>)}
+						</Select>)
+				}
+				</Grid>
+			</Grid>
+			</Grid>
+		);
+}
+
+function Compute_hcc_score(props) {
+	const classes = useButtonStyles();
+	const [sum, setSum] = React.useState(-1);
+	const [open, setOpen] = React.useState(false);
+
+	const [errDialog, setErrDialog] = React.useState({
+		show: false,
+		message: ""
+	});
+
+	function handleClose() {
+		setOpen(false);
+	}
+
+	function handleErrDialogClose() {
+		setErrDialog({show: false, message: ""});
+	}
+
+	function compute() {
+		var calc_sum = 0;
+		var err_check = 0;
+		var newErrorState = Array.from({length: props.fields_info.length}, (v, k) => false);
+
+		for (var i = 0; i < props.fields_info.length; i++) {
+			const field = props.fields_info[i];
+			const input = document.getElementById(field.name);
+
+			if (input !== null) {
+				const val = (field.type == "Float" ? parseFloat(input.value) : parseInt(input.value));
+				if (isNaN(val)) {
+					err_check = -2;
+					newErrorState[i] = true;
+				} else {
+					newErrorState[i] = false;
+				}
+				calc_sum += val * field.weight;
+			} else {
+				err_check = -1;
+				break;
+			}
+		}
+
+		if (err_check == 0) {
+			setSum(calc_sum);
+			setOpen(true);
+		} else if (err_check == -1) {
+			setErrDialog({
+				show: true,
+				message: 'Please try refreshing page (getElementById failed)'
+			});
+		}
+		// Feel free to add any other error reporting via setErrDialog
+
+		props.setErrorState(newErrorState);
+	}
+
+	return (
+		<div>
+			<Button variant="contained" color="primary" className={classes.button} onClick={compute}>Compute HCC Score</Button>
+			<Dialog
+				maxWidth='xl'
+				open={open}
+				onClose={handleClose}>
+				<DialogContent>
+					<Typography variant="overline" display="block" gutterBottom>HCC RECURRENCE SCORE</Typography>
+					<Typography variant="h1" component="h2" gutterBottom>
+						{(Math.round(sum * Math.pow(10, config.roundoff)) / Math.pow(10, config.roundoff)).toString()+"%"}
+					</Typography>
+				</DialogContent>
+			</Dialog>
+			<Dialog open={errDialog.show} onClose={handleErrDialogClose}>
+				<DialogContent>
+					<Typography variant="body1" gutterBottom>{errDialog.message}</Typography>
+				</DialogContent>
+			</Dialog>
+    	</div>);
+}
+
+function Info_button() {
+	const tooltip_classes = useBlackTooltipStyles();
+	return (
+		<Tooltip 
```

Comment:
On a phone this has to be touched for a long time before it shows up. How about this:

```suggestion
		<Tooltip 
         	    enterTouchDelay=0
```